### PR TITLE
MOON-305: POC for treeview in dropdown

### PR DESCRIPTION
**Pull request by tdraier**
## JIRA

https://jira.jahia.org/browse/MOON-305

## Description

This is POC - work in progress

### CODE REVIEW

File: src/components/Dropdown/Dropdown.scss [hunk 1]
```diff
@@ -17,6 +17,18 @@ $border-width: 1px;
     margin-right: var(--spacing-nano);
 }
 
+.moonstone-disabled {
```
Comment (ffffffelix) [inline src/components/Dropdown/Dropdown.scss:20]: It seems we can't disable an item with the `TreeViewMenu`.
- The prop `isDisabled` is not defined in the type`TreeViewData`
- Please, can you provide an example in storybook

Comment (tdraier) [reply src/components/Dropdown/Dropdown.scss:20]: - isDisabled now added
- there's already an example in storybook ( playground / tree)

File: src/components/TreeView/ControlledTreeView.tsx [hunk 2]
```diff
@@ -124,7 +126,7 @@ export const ControlledTreeView: React.FC<ControlledTreeViewProps> = ({
                             {/* TreeViewItem */}
                             <div
                                 className={clsx('flexRow_nowrap', 'alignCenter', 'flexFluid', 'moonstone-treeView_itemLabel', node.className)}
-                                onClick={handleNodeClick}
+                                onClick={!node.isDisabled && handleNodeClick}
```
Comment (ffffffelix) [inline src/components/TreeView/ControlledTreeView.tsx:129]: I have this warning
> If you used to conditionally omit it with onClick={condition && value}, pass onClick={condition ? value : undefined} instead.

File: src/components/TreeView/ControlledTreeView.tsx [hunk 3]
```diff
@@ -7,6 +7,7 @@ import Loading from '~/icons/Loading';
 import ChevronDown from '~/icons/ChevronDown';
 import ChevronRight from '~/icons/ChevronRight';
 import {Typography} from '~/components/Typography';
+import {isDisabled} from "@testing-library/user-event/dist/utils";
```
Comment (ffffffelix) [inline src/components/TreeView/ControlledTreeView.tsx:10]: Why do you import something from `testing-library` into the component?

**Review by ffffffelix — COMMENTED** [review]
I'm wondering if we should disable children when a tree node is disabled.

File: src/components/TreeView/ControlledTreeView.tsx (hunk 3 above)
Comment (tdraier) [reply src/components/TreeView/ControlledTreeView.tsx:10]: no idea :-)

**Comment by tdraier** [comment]
> I'm wondering if we should disable children when a tree node is disabled.

I think it's up to the component user to decide if he wants to disable all children, or even do not display the children at all. By just disabling the single tree node we leave the possibility to do what we want

**Comment by ffffffelix** [comment]
> > I'm wondering if we should disable children when a tree node is disabled.
> 
> I think it's up to the component user to decide if he wants to disable all children, or even do not display the children at all. By just disabling the single tree node we leave the possibility to do what we want

If we let the component user decide, we may have different behaviors for the same component and so have inconsistency in our interface. It's the role of the design system to prevent some behaviors to ensure a homogeneous experience.
We will decide later which behavior we want for this use case.

**Review by ffffffelix — APPROVED** [review]

**Pull request merged**